File: persistence/src/main/java/org/devgateway/toolkit/persistence/dto/livestock/disease/DiseaseQuantityChart.java
package org.devgateway.toolkit.persistence.dto.livestock.disease;

/**
 * @author Nadejda Mandrescu
 */
public class DiseaseQuantityChart {

    private final DiseaseQuantityConfig config;

    private final DiseaseQuantityFilter filter;

    private final DiseaseQuantityData data;

    public DiseaseQuantityChart(DiseaseQuantityConfig config, DiseaseQuantityFilter filter, DiseaseQuantityData data) {
        this.config = config;
        this.filter = filter;
        this.data = data;
    }

    public DiseaseQuantityConfig getConfig() {
        return config;
    }

    public DiseaseQuantityFilter getFilter() {
        return filter;
    }

    public DiseaseQuantityData getData() {
        return data;
    }
}

File: ui20/src/modules/entities/diseaseSituation/DiseaseQuantityChart.js
import DiseaseQuantityConfig from "./DiseaseQuantityConfig"
import DiseaseQuantityData from "./DiseaseQuantityData"
import DiseaseQuantityFilter from "./DiseaseQuantityFilter"

const DiseaseQuantityChart: {
  config: DiseaseQuantityConfig,
  filter: DiseaseQuantityFilter,
  data: DiseaseQuantityData,
} = {
  filter: DiseaseQuantityFilter
}

export default DiseaseQuantityChart

export const diseaseQuantityFromApi = ({config, filter, data} = {}) => {
  DiseaseQuantityChart.config = new DiseaseQuantityConfig(config.years)
  DiseaseQuantityChart.data = diseaseQuantityDataFromApi(data)
  DiseaseQuantityChart.filter.year = filter.year
  DiseaseQuantityChart.filter.month = DiseaseQuantityChart.data.lastMonth
  DiseaseQuantityChart.filter.diseaseId = filter.diseaseId
  return DiseaseQuantityChart
}

export const diseaseQuantityDataFromApi = ({quantities} = {}) => {
  return new DiseaseQuantityData(quantities)
}
